engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetFboWriteMask.java

```diff
@@ -49,75 +44,58 @@
      */
-    public SetFboWriteMask(boolean renderToColorBuffer, boolean renderToDepthBuffer, boolean renderToLightBuffer, ResourceUrn fboName, BaseFBOsManager fboManager) {
+    public SetFboWriteMask(FBO fbo, boolean renderToColorBuffer, boolean renderToDepthBuffer, boolean renderToLightBuffer) {
+        this.fbo = fbo;
         this.renderToColorBuffer = renderToColorBuffer;
         this.renderToDepthBuffer = renderToDepthBuffer;
         this.renderToLightBuffer = renderToLightBuffer;
-
-        this.fboName = fboName;
-        this.fboManager = fboManager;
-
-        update(); // Cheeky way to initialise fbo
-        fboManager.subscribe(this);
     }
 
     /**
      * Creates the default instance of this class for the given FBO, resetting all masks to true.
      *
-     * @param fboName A ResourceUrn identifying the FBO whose render masks have to be modified - usually only the writeOnlyGBuffer FBO.
-     * @param fboManager The FBOManager responsible for managing the given FBO.
+     * @param fbo The FBO whose render masks have to be modified - usually only the gBufferWrite FBO.
      */
-    private SetFboWriteMask(ResourceUrn fboName, BaseFBOsManager fboManager) {
+    private SetFboWriteMask(FBO fbo) {
+        this.fbo = fbo;
         this.renderToColorBuffer = true;
         this.renderToDepthBuffer = true;
         this.renderToLightBuffer = true;
 
-        this.fboName = fboName;
-        this.fboManager = fboManager;
-
-        update(); // Cheeky way to initialise fbo
-        fboManager.subscribe(this);
-
         defaultInstance = this;
     }
 
     @Override
     public StateChange getDefaultInstance() {
         if (defaultInstance == null)
-            defaultInstance = new SetFboWriteMask(fboName, fboManager);
+            defaultInstance = new SetFboWriteMask(fbo);
         return defaultInstance;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(fboName);
+        return Objects.hashCode(fbo);
     }
 
     @Override
     public boolean equals(Object obj) {
         return (obj instanceof SetFboWriteMask)
-                && fboName.equals(((SetFboWriteMask) obj).fboName)
+                && fbo.equals(((SetFboWriteMask) obj).fbo)
                 && renderToColorBuffer == ((SetFboWriteMask) obj).renderToColorBuffer
                 && renderToDepthBuffer == ((SetFboWriteMask) obj).renderToDepthBuffer
                 && renderToLightBuffer == ((SetFboWriteMask) obj).renderToLightBuffer;
     }
 
     @Override
-    public void update() {
-        fbo = fboManager.get(fboName);
-    }
-
-    @Override
     public String toString() {
-        return String.format("%30s: %s, %b, %b, %b", this.getClass().getSimpleName(), fboName, renderToColorBuffer, renderToDepthBuffer, renderToLightBuffer);
+        return String.format("%30s: %s, %b, %b, %b", this.getClass().getSimpleName(), fbo.getId(), renderToColorBuffer, renderToDepthBuffer, renderToLightBuffer);
```

Comment:
Same issue as earlier comment: the fbo name/urn is quite important for us mere mortals. =)